### DISCUSSION TITLE
Width percentage with after:: pseudo doesn't work

### DISCUSSION THREAD

**Discussion by ecemac** [tailwindlabs/tailwindcss#9167] [Help] [answered]
Hey y'all,

I love Tailwind but I can't figure this out. What I'm trying to do is, add an after::  pseudo to an element, and set it's width as a percentage.  It works with  `w-full` and `w-1/2`, but I can't get it to work with other percentages. Here's my code to clarify:

```
const getClassNames = (perc) => {
   //'perc' is a percentage, I get values like 20, 25, 50, 100 etc 
    let className = "";
    if (perc < 0) return;
    if (perc > 0) {
      className =
        "after:absolute after:bg-blue-700 after:h-full after:inset-y-0 after:left-0";
    }
    if (perc === 100) {
     //if it's 100, the content width should be 100%. This works fine
      return className + ` after:w-full`;
    } else {
     //if it's lower than 100, I calculate the fraction value and return 1/4, 1/2, 3/4 etc. Only w-1/2 works.
      return className + ` after:w-${decimalToFraction(perc / 100)}`;
    }
  };
```
<img width="342" alt="Screen Shot 2022-08-24 at 10 43 56" src="https://user-images.githubusercontent.com/43401712/186360663-0c9e9d4f-9e44-49f9-9502-5e1220a47921.png">

I can see the correct classes when I inspect the code, like `after:w-3/4`, but it doesn't apply. It seems to me that either I'm doing something wrong, or this is an issue related to after::

In any case do you guys have any solutions for me to achieve this?

**Comment by wongjn** [comment, marked as answer]
Tailwind cannot know about dynamic classes built like `after:w-${decimalToFraction(perc / 100)}` since it only scans files as text and does no language interpretation or execution. It would seem that `after:w-1/2` works only because it exists as a string somewhere else in code that Tailwind scans.

For your use case, we can look at using CSS variables, so that the `style` attribute of the element is: ``--w: `${perc}%` `` and then we can use an arbitrary value class `after:w-[var(--w)]`.
